internal/service/ec2/vpc_ipam_pool_cidr_allocation.go

```diff
@@ -191,14 +191,16 @@ func FindIpamPoolCidrAllocation(conn *ec2.EC2, id string) (*ec2.IpamPoolAllocati
 	return output.IpamPoolAllocations[0], pool_id, nil
 }
 
-func encodeIpamPoolCidrAllocationID(allocation_id, pool_id string) string {
-	return fmt.Sprintf("%s_%s", allocation_id, pool_id)
+// this function is also used in vpc_ipam_preview_next_cidr.go
+func encodeIpamPoolCidrAllocationID(general_id, pool_id string) string {
+	return fmt.Sprintf("%s_%s", general_id, pool_id)
 }
 
+// this function is also used in vpc_ipam_preview_next_cidr.go
 func DecodeIpamPoolCidrAllocationID(id string) (string, string, error) {
 	idParts := strings.Split(id, "_")
 	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
-		return "", "", fmt.Errorf("expected ID in the form of allocationId_poolId, given: %q", id)
+		return "", "", fmt.Errorf("expected ID in the form of generalId_poolId, given: %q", id)
```

Comment:
after seeing this, the error message has become too unclear for users, imo. we should either pass the values that populate the error message or revert back to functions per resource.

Reply:
I've opted for option 2 since it's a common pattern for other resources to use their own function as of the current provider design. I've reset the commits to keep the commit history for `vpc_ipam_pool_cidr_allocation.go` clean